### PR TITLE
[#537] Update participant signature for smdb

### PR DESCRIPTION
**Pull request by jhflorey**
This Pr addresses comment https://github.com/episphere/connectApp/issues/537#issuecomment-1681270458 of issue https://github.com/episphere/connectApp/issues/537

### CODE REVIEW

File: siteManagerDashboard/participantSummary.js [hunk 1]
```diff
@@ -352,17 +352,20 @@ const renderDownload = async (participant, timeStamp, fileLocation, coordinates)
     download(pdfBytes, fileLocationDownload, "application/pdf");
 }
 
-const createSignature = (participant) => {
-    const middleName = (participant[fieldMapping.consentMiddleName] !== undefined && participant[fieldMapping.consentMiddleName] !== "") ? participant[fieldMapping.consentMiddleName] : ``
+const createSignature = (participant, type) => {
+    const mddileNameCID = type === 'hipaarevoc' ? fieldMapping.HIPAARevocMiddleName : type === 'datadestruction' ? fieldMapping.dataDestructionMiddleName : fieldMapping.consentMiddleName;
```
Comment (abhinavjonnada82) [inline siteManagerDashboard/participantSummary.js:356]: Shouldn't this be 
const suffixCID = type === 'hipaarevoc' ? participant[fieldMapping.HIPAARevocSuffix] : type === 'datadestruction' ? participant[fieldMapping.dataDestructionSuffix] : participant[fieldMapping.consentSuffix];

File: siteManagerDashboard/participantSummary.js [hunk 2]
```diff
@@ -352,17 +352,20 @@ const renderDownload = async (participant, timeStamp, fileLocation, coordinates)
...
+    const middleName = (participant[mddileNameCID] !== undefined && participant[mddileNameCID] !== "") ? participant[mddileNameCID] : ``
```
Comment (abhinavjonnada82) [inline siteManagerDashboard/participantSummary.js:357]: `(participant[mddileNameCID] !== undefined && participant[mddileNameCID] !== "") ? participant[mddileNameCID] : ```

Can be simplified to` participant[mddileNameCID] || ```
Returns an empty string `` if the participant[mddileNameCID] is falsy (which includes undefined and an empty string).

File: siteManagerDashboard/participantSummary.js [hunk 3]
```diff
@@ -352,17 +352,20 @@ const renderDownload = async (participant, timeStamp, fileLocation, coordinates)
...
+const createPrintName = (participant, type) => {
     const firstName = participant[fieldMapping.consentFirstName]
-    const middleName = (participant[fieldMapping.consentMiddleName] !== undefined && participant[fieldMapping.consentMiddleName] !== "") ? participant[fieldMapping.consentMiddleName] : ``
+    const mddileNameCID = type === 'hipaarevoc' ? fieldMapping.HIPAARevocMiddleName : type === 'datadestruction' ? fieldMapping.dataDestructionMiddleName : fieldMapping.consentMiddleName;
+    const middleName = (participant[mddileNameCID] !== undefined && participant[mddileNameCID] !== "") ? participant[mddileNameCID] : ``
     const lastName = participant[fieldMapping.consentLastName]
-    const suffix = (participant[fieldMapping.consentSuffix] !== undefined && participant[fieldMapping.consentSuffix] !== "") ? fieldMapping[participant[fieldMapping.consentSuffix]] : ``
+    const suffixCID = type === 'hipaarevoc' ? fieldMapping.HIPAARevocSuffix : type === 'datadestruction' ? fieldMapping.dataDestructionSuffix : fieldMapping.consentSuffix;
+    const suffix = (participant[suffixCID] !== undefined && participant[suffixCID] !== "") ? fieldMapping[participant[suffixCID]] : ``
```
Comment (abhinavjonnada82) [inline siteManagerDashboard/participantSummary.js:368]: Same as above

**Comment by abhinavjonnada82** [comment]
Minor suggestions. Looks good!

**Review by abhinavjonnada82 — APPROVED** [review]

File: siteManagerDashboard/participantSummary.js (hunk 2 above)
Comment (jhflorey) [reply siteManagerDashboard/participantSummary.js]: Solved

File: siteManagerDashboard/participantSummary.js (hunk 3 above)
Comment (jhflorey) [reply siteManagerDashboard/participantSummary.js]: Solved

**Comment by jhflorey** [comment]
@abhinavjonnada82 please help re-check again.

File: siteManagerDashboard/participantSummary.js [hunk 4]
```diff
@@ -352,18 +352,21 @@ const renderDownload = async (participant, timeStamp, fileLocation, coordinates)
     download(pdfBytes, fileLocationDownload, "application/pdf");
 }
 
-const createSignature = (participant) => {
-    const middleName = (participant[fieldMapping.consentMiddleName] !== undefined && participant[fieldMapping.consentMiddleName] !== "") ? participant[fieldMapping.consentMiddleName] : ``
-    const createParticipantSignature = participant[fieldMapping.consentFirstName] + " " + middleName + " " + participant[fieldMapping.consentLastName]
+const createSignature = (participant, type) => {
+    const mddileNameCID = type === 'hipaarevoc' ? fieldMapping.HIPAARevocMiddleName : type === 'datadestruction' ? fieldMapping.dataDestructionMiddleName : fieldMapping.consentMiddleName;
```
Comment (anthonypetersen) [inline siteManagerDashboard/participantSummary.js:356]: please fix spelling of `middle`

**Review by anthonypetersen — CHANGES_REQUESTED** [review]
fix spelling of variable, looks fine besides that

**Review by anthonypetersen — APPROVED** [review]

**Pull request merged**